chapter2/tests/t1/chapter2-t1-two-files.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -eo pipefail
+# Args check
+if [[ "$#" -ne 2 ]]; then
+    echo "Error: Two arguments are required"
+    exit 1
+fi
+# Function for delimiter
+print_separator() {
+    printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
+}
+# Function to generate a random string and write to a file with permissions
+gen_rand_str() {
+    local file="$1"
+    openssl rand -base64 20 > "$file" && chmod 700 "$file"
+}
+
+if   [[ ! -r `pwd` ]]; then
+    echo "Directory is not readable."
+    exit 255
+elif [[ -e $1 && -e $2 ]]; then
```

Comment:
It is now better, but I still think that this solution would look much better with loops implementation - https://www.cyberciti.biz/faq/bash-for-loop/

It will help you to avoid manual check for cases when:
 - $1 exists, $2 exists
 - $1 exists, $2 not exists
 - $1 not exists, $2 exists
 - $1 not exists, $2 not exists

So you just create loop, which checks each item passed in it separately and decided what it will do with it.